hydra-main/src/main/java/com/addthis/hydra/query/aggregate/QueryTask.java

```diff
@@ -136,6 +150,29 @@ private void rescheduleSelfWithBackoff() {
         }
     }
 
+    private FrameReadTaskSourceProvider createTaskSourceProvider(MeshSourceAggregator sourceAggregator) {
+        Query query = sourceAggregator.query;
+        int totalTasks = sourceAggregator.totalTasks;
+        int maxSimul = getMaxSimul(query.getParameter("maxSimul"), totalTasks);
+        if (maxSimul == totalTasks) {
+            return new DefaultFrameReadTaskSourceProvider(sourceAggregator);
+        } else {
+            return new MaxSimulFrameReadTaskSourceProvider(sourceAggregator, maxSimul);
+        }
+    }
+
+    private int getMaxSimul(@Nullable String maxSimulQueryParam, int totalTasks) {
+        if (maxSimulQueryParam == null) {
+            return totalTasks;
+        }
+        try {
```

Comment:
I disagree. If the user provides an invalid input the query should error. Silently ignoring the maxSimul parameter violates the principle of least surprise. Throw an IllegalArgumentException it is not any more lines of code than this implementation.